### PR TITLE
ProjectAPIDao expansion 

### PR DESCRIPTION
**Pull request by realDragonium**
Perhaps its a bit too literally translated, not sure about it. for reference currently started to remake [this method ](https://github.com/SpongePowered/Ore/blob/staging/apiV2/app/db/impl/query/APIV2Queries.scala#L100) 

   
- [x] seeHidden 
- [x] categories 
- [x] pluginId 
- [x] owner
- [x] query
- [x] sort
- [x] tags 
- [x] stars and watchers (couldn't confirm it yet or it works or not)

- [x] using  StringTemplate 4

### CODE REVIEW

File: src/main/java/io/papermc/hangar/service/project/ProjectService.java [hunk 1]
```diff
@@ -53,16 +54,18 @@
     private final HangarDao<ProjectApiDao> projectApiDao;
     private final UserService userService;
     private final FlagService flagService;
+    private final Jdbi jdbi;
 
     @Autowired
-    public ProjectService(HangarConfig hangarConfig, HangarDao<ProjectDao> projectDao, HangarDao<UserDao> userDao, HangarDao<VisibilityDao> visibilityDao, HangarDao<ProjectApiDao> projectApiDao, UserService userService, FlagService flagService) {
+    public ProjectService(HangarConfig hangarConfig, HangarDao<ProjectDao> projectDao, HangarDao<UserDao> userDao, HangarDao<VisibilityDao> visibilityDao, HangarDao<ProjectApiDao> projectApiDao, UserService userService, FlagService flagService, Jdbi jdbi) {
```
Comment (MiniDigger) [inline src/main/java/io/papermc/hangar/service/project/ProjectService.java:60]: not really sure why JDBI is injected here?

**Comment by MiniDigger** [comment]
other than that it looks fine, hopefully I get some time to test later, but if anyone else with merge perms has time to check before me, go ahead

File: src/main/java/io/papermc/hangar/service/project/ProjectService.java (hunk 1 above)
Comment (realDragonium) [reply src/main/java/io/papermc/hangar/service/project/ProjectService.java]: ah, I forgot to remove it

**Comment by MiniDigger** [comment]
sorry, I butchered the rebase, merged manually

**Comment by MiniDigger** [comment]
oh and it also doesnt work, even tho I merged https://github.com/PaperMC/Hangar/pull/89 before, could you take a look at this again? I don't have time this weekend :/

**Comment by KennyTV** [comment]
Fixed in https://github.com/PaperMC/Hangar/commit/ddbd1a4d919797545945bc7cdcd5b8506e4a9d5

**Pull request closed without merging**